qiskit_nature/problems/sampling/protein_folding/peptide/chains/main_chain.py

```diff
@@ -105,57 +92,30 @@ def _build_main_chain(
             )
             main_chain.append(main_bead)
-        main_bead = MainBead(main_chain_len - 1, None, None, None)
+        main_bead = MainBead(main_chain_len - 1, "", None, None)
         main_chain.append(main_bead)
         return main_chain
 
     @staticmethod
     def _validate_side_chain_lengths(
         main_chain_len: int,
-        side_chain_lens: List[int],
-        side_chain_residue_sequences: List[Optional[str]],
+        side_chain_residue_sequences: List[str],
     ) -> None:
-        if side_chain_lens is not None and main_chain_len != len(side_chain_lens):
-            raise InvalidSizeException(
-                f"The length of list of side chain lengths provided: {len(side_chain_lens)}, "
-                f"does not equal the length of the main chain: {main_chain_len}"
-            )
-        if side_chain_lens is not None and main_chain_len != len(side_chain_residue_sequences):
+        if main_chain_len != len(side_chain_residue_sequences):
             raise InvalidSizeException(
                 f"The length of list of main chain residue sequences: "
                 f"{len(side_chain_residue_sequences)} does not equal the length of the main "
                 f"chain: {main_chain_len}"
             )
 
     @staticmethod
-    def _validate_main_chain_lengths(
-        main_chain_len: int, main_chain_residue_sequence: List[str]
-    ) -> None:
-        if main_chain_len != len(main_chain_residue_sequence):
-            raise InvalidSizeException(
-                f"The length of list of main chain residue sequences: "
-                f"{len(main_chain_residue_sequence)} does not equal the length of the main "
-                f"chain: {main_chain_len}"
-            )
-
-    @staticmethod
-    def _validate_side_chain_index_by_lengths(side_chain_lens: List[int]) -> None:
-        if side_chain_lens is not None and (
-            side_chain_lens[0] != 0 or side_chain_lens[1] != 0 or side_chain_lens[-1] != 0
-        ):
-            raise InvalidSideChainException(
-                "First, second and last main beads are not allowed to have a side chain. Non-zero "
-                "length provided for an invalid side chain."
-            )
-
-    @staticmethod
-    def _validate_side_chain_index_by_residues(
-        side_chain_residue_sequences: List[Optional[str]],
+    def _validate_side_chain_index(
+        side_chain_residue_sequences: List[str],
     ) -> None:
         if side_chain_residue_sequences is not None and (
-            side_chain_residue_sequences[0] is not None
-            or side_chain_residue_sequences[1] is not None
-            or side_chain_residue_sequences[-1] is not None
+            side_chain_residue_sequences[0] != ""
+            or side_chain_residue_sequences[1] != ""
+            or side_chain_residue_sequences[-1] != ""
         ):
             raise InvalidSideChainException(
                 "First, second and last main beads are not allowed to have a side chain. Non-None "
```

Comment:
Is this limitation of 1st, 2nd and last not being permitted a side chain exposed when you create a Peptide on its constructor. If I follow things down it seems that it just builds a MainChain and will fail if this is not the case. Looking at the example code, I guess I assumed that the absence of side chains in those places was just because it was an example, not that it was required as such. Is this something likely to change in the future? If not why would we want to user to pass these - would it make sense to define the side chain sequences as things that can be set - i.e length main chain - 3 so the first side chain in the sequence applies to main bead 3 (assuming indexes start at 1). Maybe passing them is clearer in that then each main has a side chain entry. But I think the behavior of 1st/2nd and last needs to be surfaced.

Reply:
There are multiple aspects here:

1. 1st, 2nd and last not allowed: this is clearly documented in the Peptide's docstrings. The docstrings states that "this can be done without loss of generality" and refers to the paper for more information.

2. changing the side chain sequence: I don't think, that this is a good idea. Simply in terms of usability, I find it difficult to justify why a list `["a", "b", "c"]` would attach to some main sequence `"ABCDEF"` at the positions `a -> C`, `b -> D` and `c -> E`. That is both, not easy to document or explain and also just confusing. I would then rather see a good warning for the user.

3. Unittests: I am not sure if this exists already, but arguably there should be some unittests which assert the correct warnings being raised in edge cases similar to this.

Reply:
I agree with Max and that was my motivation. 

Not crucial to this discussion but there is an issue to allow side chains on the second main bead at an expense of an additional qubit required for encoding the problem (#252) 

Thanks to Max I realized that my docs regarding this need to be fixed. I made too big of a shorthand in writing that actually result in a false claim.

Reply:
We do have unit tests for that:
```
    def test_main_chain_illegal_side_chain_first(self):
        """Tests that an exception is thrown in case of illegal side chain."""
        main_chain_residue_seq = "SAAA"
        side_chain_residue_sequences = ["A", "", "S", ""]
        with self.assertRaises(InvalidSideChainException):
            _ = MainChain(
                main_chain_residue_seq,
                side_chain_residue_sequences,
            )

    def test_main_chain_illegal_side_chain_second(self):
        """Tests that an exception is thrown in case of illegal side chain."""
        main_chain_residue_seq = "SAAA"
        side_chain_residue_sequences = ["", "A", "S", ""]
        with self.assertRaises(InvalidSideChainException):
            _ = MainChain(
                main_chain_residue_seq,
                side_chain_residue_sequences,
            )

    def test_main_chain_illegal_side_chain_last(self):
        """Tests that an exception is thrown in case of illegal side chain."""
        main_chain_residue_seq = "SAAA"
        side_chain_residue_sequences = ["", "", "A", "S"]
        with self.assertRaises(InvalidSideChainException):
            _ = MainChain(
                main_chain_residue_seq,
                side_chain_residue_sequences,
            )
```

Reply:
>  1st, 2nd and last not allowed: this is clearly documented in the Peptide's docstrings. The docstrings states that "this can be done without loss of generality" and refers to the paper for more information.

Where are you looking to see that - this is what I see in Peptide where I do not see the text you refer to
```
class Peptide:
    """A class containing information about beads and chains of a protein."""

    def __init__(
        self,
        main_chain_residue_sequence: str,
        side_chain_residue_sequences: List[str],
    ):
        """
        Args:
            main_chain_residue_sequence: String of characters that define residues for a main
                                        chain. Valid residue types are [A, C, D, E, F, G, H, I,
                                        K, L, M, N, P, Q, R, S, T, V, W, Y].
            side_chain_residue_sequences: List of characters that define residues for all side
                                        beads. Empty string if a side bead does not exist. Valid
                                        residue types are [A, C, D, E, F, G, H, I, K, L, M, N, P,
                                        Q, R, S, T, V, W, Y].
        """
```

Reply:
Sorry, it was indeed not in the Peptide but in the module-level docstring: https://github.com/Qiskit/qiskit-nature/blob/main/qiskit_nature/problems/sampling/protein_folding/__init__.py#L16-L18

Reply:
Update: I found the text you refer to - it occurred to me after writing the above it might be in the overview text in the init file - and yes it was here qiskit_nature/problems/sampling/protein_folding/__init__.py

I still think PepTide class itself thought should make this clear when going to its API ref.

Reply:
Sounds good! @dlasecki could you copy that into the docstring of Peptide?

Reply:
That comment sequence just above looks weird now - I tell you I figured it out after told me where it was! I had left the page open and perhaps it had not refreshed when I added the update after the prior comment - oh well!

Reply:
Do you mean duplicating what's in the init file to the Peptide docstring? Not sure about it... Maybe we could split it somehow and avoid duplication...

Reply:
I agree with Steve that this needs to go into the `Peptide` docstring. Whether we duplicate it on the module level or not is fine either way to me

Reply:
If the more descriptive overall text stays in init, that would seem fine to me. I think here it should at least note the restriction on 1st, 2nd and last and perhaps link back to the main descriptive text for a higher level view/more information overall. It does not have to duplicate the entire text but there should be at least enough in PepTide to make them aware of the restriction/behavior if they first arrive at this class.

Reply:
Ah I did not want to say to copy the entire module level docstring. I would have only copied the sentence referring to the side chain restrictions.

Reply:
Updated the docs.